pkg/controller/elasticsearch/settings/environment.go

```diff
@@ -13,8 +13,9 @@ const (
 	EnvReadinessProbeProtocol = "READINESS_PROBE_PROTOCOL"
 	HeadlessServiceName       = "HEADLESS_SERVICE_NAME"
 
-	// EnvPodName and EnvPodIP are injected as env var into the ES pod at runtime,
+	// These are injected as env var into the ES pod at runtime,
 	// to be referenced in ES configuration file
-	EnvPodName = "POD_NAME"
-	EnvPodIP   = "POD_IP"
+	EnvPodName  = "POD_NAME"
+	EnvPodIP    = "POD_IP"
+	EnvNodeName = "node.attr.NODE_NAME"
```

Comment:
I'm not a big fan of env variables with dots as it is not supported by bash (which is the default shell in the ES image).
At first you may think that the env variable is empty if you use `export` :

```
bash-1.2$ export
...
declare -x READINESS_PROBE_PROTOCOL="https"
declare -x SHLVL="1"
declare -x TERM="xterm"
declare -x node.attr.NODE_NAME # empty ?
bash-4.2$
```

Than you will hit this kind of error if you try to print it:

```
bash-4.2$ echo ${node.attr.NODE_NAME}
bash: ${node.attr.NODE_NAME}: bad substitution
```

Eventually you'll find that you need something like awk to print the value:

```
bash-4.2$ awk 'BEGIN {print ENVIRON["node.attr.NODE_NAME"]}'
gke-michael-dev-2-default-pool-1cc419ff-nl3d
```

Is there a disadvantage to setting this value in the configuration file ?
I think your proposal does the job, so I'm also fine to keep it as is.

Reply:
No that's a good point. I just had tunnel vision after I saw that ES does not do the `_` -> `.` conversion like Kibana does. Fixing